server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -23,16 +26,190 @@
     /**
      * Clusters the results from the query.
      *
-     * TODO: Currently returns the same results i.e. method is a passthrough right now
-     *
      * @param tabularQueryResult the result of the unclustered query
      * @return the new clustered results
      */
     public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
+        if (this.clusterClause.getFieldType().equals("number")) {
+            return this.aggregateNumber(tabularQueryResult);
+        }
+
         this.clusterClause = null; // reset cluster clause
         return tabularQueryResult;
     }
 
+    /**
+     * Clusters results by number aggregation.
+     *
+     * @param tabularQueryResult the original results
+     * @return the tabularqueryresult aggregated by numbers with the specifications given by the cluster clause
+     */
+    private TabularQueryResult aggregateNumber(TabularQueryResult tabularQueryResult) {
+        List<List<Object>> clusters = this.clusterClause.getClusters();
+
+        // establish keys
+        String fieldNameKey = this.clusterClause.getFieldNames().get(0);
+        String aggregationNameKey = this.clusterClause.getAggregationName();
+
+        List<Map<String, Object>> data = tabularQueryResult.getData();
+
+        // determine ordering
+        Map<String, Object> first = data.get(0);
+        Map<String, Object> last = data.get(data.size() - 1);
+        BigDecimal firstGroup = new BigDecimal(first.get(fieldNameKey).toString()); // assume number because 'number' fieldType
+        BigDecimal lastGroup = new BigDecimal(last.get(fieldNameKey).toString());
+        String order = "asc";
+        if (firstGroup.compareTo(lastGroup) == 1) {
+            order = "des";
+        }
```

Comment:
Not a big deal, but this block could be condensed with a ternary operator.

`String order = firstGroup.compareTo(lastGroup) == 1 ? "des" : "asc";`

Same with the cluster count below.

Reply:
Fixed.